apps/storybook/src/Dialog.stories.tsx

```diff
@@ -0,0 +1,272 @@
+            </Button>
+            <Button onClick={secondaryButtonHandle}>Secondary</Button>
+          </Dialog.ButtonBar>
+        </Dialog.Main>
+      </Dialog>
+    </>
+  );
+};
+
+export const Draggable: Story = () => {
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  const closeDialog = () => {
+    setIsOpen(false);
+  };
+
+  const onClose = (event: React.SyntheticEvent<Element, Event>) => {
+    action('onClose', { depth: 1 })(event);
+    closeDialog();
+  };
+
+  const primaryButtonHandle = (
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+  ) => {
+    action('Primary button')(event);
+    closeDialog();
+  };
+
+  const secondaryButtonHandle = (
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+  ) => {
+    action('Secondary button')(event);
+    closeDialog();
+  };
+
+  return (
+    <>
+      <Button styleType='high-visibility' onClick={() => setIsOpen(true)}>
+        Open Dialog
+      </Button>
+      <Dialog
+        isOpen={isOpen}
+        onClose={onClose}
+        closeOnEsc
+        isDismissible
+        isDraggable
+      >
+        <Dialog.Main>
+          <Dialog.TitleBar titleText='Best dialog ever' />
+          <Dialog.Content>
+            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
+            eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim
+            ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
+            aliquip ex ea commodo consequat. Duis aute irure dolor in
+            reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla
+            pariatur. Excepteur sint occaecat cupidatat non proident, sunt in
+            culpa qui officia deserunt mollit anim id est laborum.
+          </Dialog.Content>
+          <Dialog.ButtonBar>
+            <Button styleType='high-visibility' onClick={primaryButtonHandle}>
+              Primary
+            </Button>
+            <Button onClick={secondaryButtonHandle}>Secondary</Button>
+          </Dialog.ButtonBar>
+        </Dialog.Main>
+      </Dialog>
+    </>
+  );
+};
+
+export const DraggableRelativeToContainer: Story = ({ isOpen, ...rest }) => {
+  const [isDialogOpen, setIsDialogOpen] = React.useState(isOpen);
+
+  React.useEffect(() => {
+    setIsDialogOpen(isOpen);
+  }, [isOpen]);
+
+  const closeDialog = () => {
+    setIsDialogOpen(false);
+  };
+
+  const onClose = (event: React.SyntheticEvent<Element, Event>) => {
+    action('onClose', { depth: 1 })(event);
+    closeDialog();
+  };
+
+  const primaryButtonHandle = (
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+  ) => {
+    action('Primary button')(event);
+    closeDialog();
+  };
+
+  const secondaryButtonHandle = (
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+  ) => {
+    action('Secondary button')(event);
+    closeDialog();
+  };
+
+  return (
+    <>
+      <Button styleType='high-visibility' onClick={() => setIsDialogOpen(true)}>
```

Comment:
Maybe put this button inside the custom div so that it's obvious that the dialog-wrapper will overlay the user content.
![](https://user-images.githubusercontent.com/9084735/188235690-fdb37664-419e-4657-b259-6eec0a7edb09.png)

Could also add some text explaining that the user needs to specify `position: relative`.

Reply:
Changed